crates/fuel-core/src/state/rocks_db.rs

```diff
@@ -712,20 +748,36 @@ where
     }
 
     fn get(&self, key: &[u8], column: Self::Column) -> StorageResult<Option<Value>> {
-        self.metrics.read_meter.inc();
-        let column_metrics = self.metrics.columns_read_statistic.get(&column.id());
-        column_metrics.map(|metric| metric.inc());
-
-        let value = self
+        let result = self
             .db
             .get_cf_opt(&self.cf(column), key, &self.read_options)
-            .map_err(|e| DatabaseError::Other(e.into()))?;
+            .map_err(|e| StorageError::Other(DatabaseError::Other(e.into()).into()));
 
-        if let Some(value) = &value {
-            self.metrics.bytes_read.inc_by(value.len() as u64);
-        }
+        self.register_read(result, column.id())
+            .map(|opt| opt.map(Arc::new))
+    }
+
+    fn get_batch<'a>(
```

Comment:
We already have `multi_get` function that you can use, you don't need to add a new one=)

Reply:
Oh right. One problem with the existing function is that it returns a `Result<Vec<...>>` whereas our use cases expect a `Vec<Result<...>>`. This should be easy to refactor as I just need to remove the `try_collect` and map the error, so I'll do that.

Reply:
Done in https://github.com/FuelLabs/fuel-core/pull/2396/commits/3d0a71f59870fc787e1e85d16ce8fcb76864054e